From 857b536507eb4561532f64dd220529218c50637b Mon Sep 17 00:00:00 2001
From: Paul Holzinger <pholzing@redhat.com>
Date: Fri, 11 Apr 2025 18:34:42 +0200
Subject: [PATCH] test/system: add prefetch users to use cache image

When using a custom --root it will not have the image present and as
such cause a pull. We can however use our own local cache if present to
avoid the pull if we give the right podman options via
_PODMAN_TEST_OPTS.

I saw the volume quota test fail during the pull in openQA thus I
noticed this issue.

Signed-off-by: Paul Holzinger <pholzing@redhat.com>
---
 test/system/161-volume-quotas.bats | 9 ++++++++-
 test/system/520-checkpoint.bats    | 8 +++++---
 2 files changed, 13 insertions(+), 4 deletions(-)

diff --git a/test/system/161-volume-quotas.bats b/test/system/161-volume-quotas.bats
index 8019499509..6fcf4dfd00 100644
--- a/test/system/161-volume-quotas.bats
+++ b/test/system/161-volume-quotas.bats
@@ -61,8 +61,15 @@ function teardown() {
     vol_two="testvol2"
     run_podman $safe_opts volume create --opt o=size=4m $vol_two
 
+    # prefetch image to avoid registry pulls because this is using a
+    # unique root which does not have the image already present.
+    # _PODMAN_TEST_OPTS is used to overwrite the podman options to
+    # make the function aware of the custom --root.
+    _PODMAN_TEST_OPTS="$safe_opts --storage-driver $(podman_storage_driver)" _prefetch $IMAGE
+
     ctrname="testctr"
-    run_podman $safe_opts run -d --name=$ctrname -i -v $vol_one:/one -v $vol_two:/two $IMAGE top
+    # pull never to ensure the prefetch works correctly
+    run_podman $safe_opts run -d --pull=never --name=$ctrname -i -v $vol_one:/one -v $vol_two:/two $IMAGE top
 
     run_podman $safe_opts exec $ctrname dd if=/dev/zero of=/one/oneMB bs=1M count=1
     run_podman 1 $safe_opts exec $ctrname dd if=/dev/zero of=/one/twoMB bs=1M count=1
diff --git a/test/system/520-checkpoint.bats b/test/system/520-checkpoint.bats
index 5113f6785e..a581f1bf47 100644
--- a/test/system/520-checkpoint.bats
+++ b/test/system/520-checkpoint.bats
@@ -134,10 +134,12 @@ function setup() {
 @test "podman checkpoint --export, with volumes" {
     skip_if_remote "Test uses --root/--runroot, which are N/A over remote"
 
-    # To avoid network pull, copy $IMAGE straight to temp root
     local p_opts="$(podman_isolation_opts ${PODMAN_TMPDIR}) --events-backend file"
-    run_podman         save -o $PODMAN_TMPDIR/image.tar $IMAGE
-    run_podman $p_opts load -i $PODMAN_TMPDIR/image.tar
+    # prefetch image to avoid registry pulls because this is using a
+    # unique root which does not have the image already present.
+    # _PODMAN_TEST_OPTS is used to overwrite the podman options to
+    # make the function aware of the custom --root.
+    _PODMAN_TEST_OPTS="$p_opts --storage-driver $(podman_storage_driver)" _prefetch $IMAGE
 
     # Create a volume, find unused network port, and create a webserv container
     volname=v-$(safename)
